opengrok-indexer/src/test/java/org/opengrok/indexer/web/StatisticsTest.java

```diff
@@ -203,35 +201,48 @@ public void testAddRequestMultipleTimeSum() {
      * Test of toJson method, of class Statistics.
      */
     @Test
-    public void testToJsonOnInstance() {
-        checkToJson(new Function<Statistics, JSONObject>() {
+    public void testToJsonOnInstance() throws IOException {
+        checkToJson(new Function<Statistics, String>() {
             @Override
-            public JSONObject apply(Statistics t) {
-                return t.toJson();
+            public String apply(Statistics t) {
+                try {
+                    return t.toJson();
+                } catch (IOException e) {
+                    e.printStackTrace();
+                }
+                return null;
             }
         });
     }
 
     /**
-     * Test of from method, of class Statistics.
+     * Test of fromJson method, of class Statistics.
      */
     @Test
     @SuppressWarnings("unchecked")
-    public void testFrom() {
-        JSONObject input = new JSONObject();
-        input.put(Statistics.STATISTIC_MINUTES, 10L);
-        input.put(Statistics.STATISTIC_REQUESTS, 100L);
-        input.put(Statistics.STATISTIC_REQUESTS_PER_MINUTE, 543L);
-        input.put(Statistics.STATISTIC_REQUESTS_PER_MINUTE_AVG, 54312.0);
-        input.put(Statistics.STATISTIC_REQUESTS_PER_MINUTE_MIN, 49L);
-        input.put(Statistics.STATISTIC_REQUESTS_PER_MINUTE_MAX, 4753L);
-        Statistics stat = Statistics.from(input);
+    public void testFrom() throws IOException {
+        ByteArrayOutputStream stream = new ByteArrayOutputStream();
+        JsonFactory jfactory = new JsonFactory();
+        JsonGenerator jGenerator = jfactory
```

Comment:
what's the point of using generator when you could just write simple json string as a string?

Reply:
The advantage is that the other fields do not have to be specified - this also tests the parsing of not fully formed Statistics JSON string.

Reply:
I like to have to construct the JSON string programmatically, it is less error prone.